### PR TITLE
refactored seo helpers and no indexed category and author pages

### PR DESCRIPTION
**Pull request by matthewl20**

### CODE REVIEW

**Comment by matthewl20** [comment]
Refactored SEO Helpers to fix rubocop offenses 

No Indexed both the category and the author pages.  I also fixed the issue where we would include both noindex and a canonical url meta tags.  Now if an article/page is set to noindex the canonical url won't be added even if one is set on the specific page.

**Comment by matthewl20** [comment]
Key components to review: indexable?, seo_canonical, seo_robots

File: app/helpers/seo_helper.rb [hunk 1]
```diff
@@ -77,7 +76,23 @@ def whitelisted_params
...
   end
 
+  private
+
+  NOINDEX_KEYS = %w[utf8 q topic].freeze
+  NOINDEX_ROUTES = {
+    authors: [:show],
+    categories: [:show]
+  }.freeze
+
```
Comment (rift137) [inline app/helpers/seo_helper.rb]: Comment this baddie

**Comment by matthewl20** [comment]
After doing more research, I chose to only noindex author pages and left category pages alone.  I read a few posts that suggested that is a better approach. (https://yoast.com/category-seo/).  But there are some things that we can do to improve those pages.

**Pull request merged**